### PR TITLE
(CONT-850) Puppet 8 compatibility

### PR DESCRIPTION
**Pull request by LukasAud**
This PR aims to update the module with compatibility for the upcoming
Puppet 8 agent release. The most relevant changes introduced in this
updated are:
Updated version requirements in metadata.json
Bumped Rubocop gems
Adjusted Rubocop configuration
Resolved new Rubocop offenses

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ fixtures:
     facts: 'https://github.com/puppetlabs/puppetlabs-facts.git'
     puppet_agent:
       repo: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
-      ref: v4.12.1
+      ref:  v4.13.0
```
Comment (jordanbreen28) [inline .fixtures.yml:12]: small bit of whitespace

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - 'spec/defines/accounts_user_spec.rb'
+    - 'spec/functions/accounts_ssh_authorized_keys_line_parser_spec.rb'
+
+# Offense count: 29
+# Configuration parameters: AllowedGroups.
+RSpec/NestedGroups:
+  Max: 5
+
+# Offense count: 1
+Style/MixinUsage:
```
Comment (jordanbreen28) [inline .rubocop_todo.yml:70]: This seems to be a recurring cop violation in spec_helper.
I seen in other PRs this was disabled inline, do you think we should do the same here? Just curious

Comment (LukasAud) [reply .rubocop_todo.yml:70]: Yeah, could do that.

File: .fixtures.yml (hunk 1 above)
Comment (LukasAud) [reply .fixtures.yml:12]: Will fix

**Review by jordanbreen28 — APPROVED** [review]

**Pull request merged**